nosepipe.py

```diff
@@ -191,15 +195,31 @@ def __init__(self):
         nose.plugins.Plugin.__init__(self)
         self._test = None
         self._test_proxy = None
-        self._argv = [os.path.abspath(sys.argv[0]),
-                      '--with-process-isolation-reporter']
-        self._argv += ProcessIsolationPlugin._get_nose_whitelisted_argv()
+        nosearg = None
+        for i in range(0, len(sys.argv)):
+            if 'nosetests' in sys.argv[i]:
+                self._argv = [sys.argv[i]]
+                nosearg = i
+                break
+
+        if nosearg is None:
+            raise Exception("nosetests not found in command-line")
+
+        # self._argv = [os.path.abspath(sys.argv[0])]
+        # if 'nosetests' not in sys.argv[0]:
+        #     for i in range(1, len(sys.argv)):
+        #         self._argv += [sys.argv[i]]
+        #         if 'nosetests' in sys.argv[i]:
+        #             break
+
+        self._argv += ['--with-process-isolation-reporter']
+        self._argv += ProcessIsolationPlugin._get_nose_whitelisted_argv(nosearg=nosearg)
         # Getting cwd inside SubprocessTestProxy.__call__ is too late - it is
         # already changed by nose
         self._cwd = os.getcwd()
 
     @staticmethod
-    def _get_nose_whitelisted_argv():
+    def _get_nose_whitelisted_argv(nosearg=0):
```

Comment:
Could you rename this to something with more meaning in this function - perhaps "offset"? I'd also like to see the +1 added to the argument before passing it in, rather than doing it multiple times inside of here.